### PR TITLE
feat: adds validation to cover start date

### PR DESCRIPTION
**Pull request by pbradley-f4**

### CODE REVIEW

File: gef-ui/server/controllers/about-facility/index.js [hunk 1]
```diff
@@ -55,30 +57,53 @@ const validateAboutFacility = async (req, res) => {
...
+          });
+        }
+      }
+
+      if (!coverEndDateDay || !coverEndDateMonth || !coverEndDateYear) {
+        aboutFacilityErrors.push({
+          errRef: 'coverEndDate',
+          errMsg: 'Enter a cover end date',
+        });
+      }
```
Comment (ttbarnes) [inline gef-ui/server/controllers/about-facility/index.js:106]: I think need to add some validation for day/month/year missing, for GDS. i.e "X must include a year": https://design-system.service.gov.uk/components/date-input/#dates-with-errors-example

File: e2e-tests/gef/cypress/integration/about-facility.spec.js [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
 /* eslint-disable no-undef */
+import moment from 'moment';
 import relative from './relativeURL';
 import aboutFacility from './pages/about-facility';
 import CREDENTIALS from '../fixtures/credentials.json';
 
 const applications = [];
 let token;
 
+const now = moment();
```
Comment (ttbarnes) [inline e2e-tests/gef/cypress/integration/about-facility.spec.js:11]: should we be using date-fns for this?

File: gef-ui/server/controllers/about-facility/index.js (hunk 1 above)
Comment (pbradley-f4) [reply gef-ui/server/controllers/about-facility/index.js:106]: Ive added this

File: e2e-tests/gef/cypress/integration/about-facility.spec.js (hunk 2 above)
Comment (pbradley-f4) [reply e2e-tests/gef/cypress/integration/about-facility.spec.js:11]: Im leaving the cypress piece for now as it uses moment everywhere

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**